mysql-test/main/merge.result

```diff
@@ -3445,6 +3447,8 @@ c1	c2
 411	421
 511	521
 #
+# MDEV-23485 - Change table to merge engine may cause table data lost.
+truncate table m1;
```

Comment:
its not necessary to comment this line or the above truncate table. Its covered in the commit message and could distract a reader to think this is part of the MDEV-23485 test .

The test block below is good but you'll notice in the results that the error message isn't that useful for a user (hence review requesting another error message).